node_group_advanced.tf

```diff
@@ -82,6 +82,13 @@ locals {
 
 # One launch template per node group
 resource "aws_launch_template" "quortex_launch_tpl" {
+
+  lifecycle {
+    precondition {
+      condition     = length(each.value.instance_types) > 0
+      error_message = "node group '${each.key}' must have at least one instance type defined"
+    }
+  }
```

Comment:
Lifecycle are usually in the bottom of the resources, could you move it there ? 🙏